### ISSUE TITLE
[Accessibility]: Focus ring visibility issues with Notepad

### ISSUE THREAD

**Issue by fkling** [sourcegraph/sourcegraph#35159]
### Audit type

Keyboard navigation

### User journey audit issue

https://github.com/sourcegraph/sourcegraph/issues/34198

### Problem description

1. Enable notepad by going to https://sourcegraph.com/notebooks and click the "Enable Notepad" button (notepad should be visible in the bottom right corner).
2. Perform a search on https://sourcegraph.com/search that yields results.
3. Add the result at least once to the notepad.
4. Focus to the notepad.
5. Move focus back and forth (via tab / shift+tab).

There a couple of elements for which the focus ring is not completely shown.

- [ ] Header button (open and closed): Focus ring not visible
![notepad-focusring-header-open](https://user-images.githubusercontent.com/179026/167489380-8af25962-44da-4c8e-8fe8-7be1f508e1a3.png)
![2022-05-09_21-25](https://user-images.githubusercontent.com/179026/167489381-961d430e-011b-412a-906b-2234451d7e32.png)
- [ ] Entries: Only the bottom part of the focus ring is visible for the first entry, and the top and bottom part for all other entries.
![2022-05-09_16-11](https://user-images.githubusercontent.com/179026/167489840-b628f31e-df8b-42ab-a3ce-270acb08efb6.png)

### Expected behavior

Focus ring is visible.

### Additional details

The main reason for this behavior is the `overflow: hidden` on the root element.

### Assigning labels

- [ ] Please give this issue an estimate by applying a label like `estimate/Xd`, where X is the estimated number of days it will take to complete.
- [X] If this issue is specific to a specific Sourcegraph product, please assign the appropriate team label to this issue.
- [ ] If this issue will require input from designers in order to complete, please assign the label `needs-design`.
- [ ] If you are confident that this issue should be fixed by GitStart, please assign the label `gitstart`.

### Owner

This issue will be fixed by my team, I have assigned a relevant member to it, or I will do so in the near future.

**Comment by sourcegraph-bot-2** [comment]
Heads up @benvenker @lguychard - the "team/search-product" label was applied to this issue.

**Comment by limitedmage** [comment]
This is happening because we have `overflow: auto` on the notepad so it scrolls, and we implement focus rings using `box-shadow`, which can be cut off by overflow (unlike the default of using `outline`). @sourcegraph/frontend-platform I'm assigning this to you as this would affect any other focusable items inside the app. We should reconsider using `box-shadow` for focus outlines and try using `outline` instead. Alternatively, `filter: drop-shadow()` might not be cut off by outlines.

**Comment by sourcegraph-bot-2** [comment]
Heads up @taylorsperry @jasongornall - the "team/frontend-platform" label was applied to this issue.

**Comment by umpox** [comment]
Thanks @limitedmage I'll take a look, I think we originally used box-shadow because we needed an offset, and outline-offset wasn't supported by all browsers at the time. I think it is now, so we might be able to easily transition to that

**Comment by camdencheek** [comment]
Related to #44878
